### PR TITLE
Add a zoekt module

### PR DESCRIPTION
**Pull request by achew22**
This module is a basic way of interacting with Zoekt

### CODE REVIEW

File: libs/zoekt/src/lib/zoekt.component.html [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<div>
+  <mat-card *ngFor="let file of (searchResult$ | async).Files">
+    <div>Repo: {{ file.Repo }}</div>
+    <div>Branches: {{ file.Branches }}</div>
+    <div>FileName: <a routerLink="/kythe.io/{{ file.FileName }}">{{ file.FileName }}</a></div>
+    <ul>
+      <li *ngFor="let line of file.Lines">
```
Comment (mrmeku) [inline libs/zoekt/src/lib/zoekt.component.html]: Consider adding a trackBy function here (we'll probably want one later)

File: libs/zoekt/src/lib/zoekt.component.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<div>
+  <mat-card *ngFor="let file of (searchResult$ | async).Files">
+    <div>Repo: {{ file.Repo }}</div>
+    <div>Branches: {{ file.Branches }}</div>
+    <div>FileName: <a routerLink="/kythe.io/{{ file.FileName }}">{{ file.FileName }}</a></div>
```
Comment (mrmeku) [inline libs/zoekt/src/lib/zoekt.component.html]: [routerLink]="'/kythe.io/' + file.FileName"

File: libs/zoekt/src/lib/zoekt.module.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  Route,
+  RouterModule,
+  UrlMatchResult,
+  UrlSegment
+} from '@angular/router';
+
+import { ZoektComponent } from './zoekt.component';
+import { ZoektService } from './zoekt.service';
+
+export const routes: Route[] = [{ path: 'search', component: ZoektComponent }];
```
Comment (mrmeku) [inline libs/zoekt/src/lib/zoekt.module.ts]: The importing module already search that /search should match this module. I believe that this path should be ''

File: libs/zoekt/src/lib/zoekt.service.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+import { ZoektSearchRequest, ZoektSearchResponse } from './zoekt.declarations';
+
+const baseUrl = 'https://cors-anywhere.herokuapp.com/zoekt.achew22.com';
+
+@Injectable({ providedIn: 'root' })
+export class ZoektService {
+  search(zoektSearchRequest: ZoektSearchRequest) {
+    return this.http.post<ZoektSearchResponse>(
+      baseUrl + '/api/search',
```
Comment (mrmeku) [inline libs/zoekt/src/lib/zoekt.service.ts]: template string?

File: libs/zoekt/src/lib/zoekt.service.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+      zoektSearchRequest,
+      { headers: { 'Content-Type': 'application/json; charset=utf-8' } }
```
Comment (mrmeku) [inline libs/zoekt/src/lib/zoekt.service.ts]: I think this header is set automatically? Am I wrong

Comment (achew22) [reply libs/zoekt/src/lib/zoekt.service.ts]: It needs to have the charset=utf-8 at the end. The default doesn't transmit that and the rest API is really picky about it.

**Pull request merged**